pinot-query-planner/src/main/java/org/apache/pinot/query/planner/plannode/PlanNodeVisitor.java

```diff
@@ -63,4 +63,150 @@ public interface PlanNodeVisitor<T, C> {
   T visitExchange(ExchangeNode exchangeNode, C context);
 
   T visitExplained(ExplainedNode node, C context);
+
+  /**
```

Comment:
Because it is closely related to this visitor. Also I don't need to call it `PlanNodeDepthFirstVisitor` given it is already contained inside `PlanNodeVisitor`. But I'm open to move as a regular class if it is preferred.